discovery/manager.go

```diff
@@ -137,7 +143,26 @@ func (m *Manager) runProvider(ctx context.Context, poolKey poolKey, updates chan
 				return
 			}
 			m.updateGroup(poolKey, tgs)
-			m.syncCh <- m.allGroups()
+			m.recentlyUpdatedMtx.Lock()
+			m.recentlyUpdated = true
+			m.recentlyUpdatedMtx.Unlock()
```

Comment:
Why not just put the following after `m.syncCh <- ...` here:

```go
select {
case <-time.After(5 * time.Second):
case <-ctx.Done():
    return
}
```

Reply:
The solution here will send updates even if there were none.

Reply:
We tried this initially, but this just blocks the sender and the updates just pile up and get delayed.

What we want is to drain the sender , but send updates to the Scraper only once at a given interval.

Reply:
@fabxc Not sure why this will send updates even if none exist though.

The issue with doing a `time.After(5 * time.Second)` is that if the SD mechanism launches a goroutine per update, we might end up with a lot of waiting goroutines because we are not draining the channel.